tests/strategy/models/energy_parameters/test_energy_params_eb.py

```diff
@@ -0,0 +1,473 @@
+                 time_slot=pendulum.datetime(2022, 2, 1, 20, 30)),
+            call(purchased_energy_Wh=0.0, area_name=area_name,
+                 time_slot=pendulum.datetime(2022, 2, 1, 20, 45)),
+            call(purchased_energy_Wh=0.0, area_name=area_name,
+                 time_slot=pendulum.datetime(2022, 2, 1, 21, 0)),
+            call(purchased_energy_Wh=0.0, area_name=area_name,
+                 time_slot=pendulum.datetime(2022, 2, 1, 21, 15)),
+            call(purchased_energy_Wh=0.0, area_name=area_name,
+                 time_slot=pendulum.datetime(2022, 2, 1, 21, 30)),
+            call(purchased_energy_Wh=0.0, area_name=area_name,
+                 time_slot=pendulum.datetime(2022, 2, 1, 21, 45)),
+            call(purchased_energy_Wh=0.0, area_name=area_name,
+                 time_slot=pendulum.datetime(2022, 2, 1, 22, 0)),
+            call(purchased_energy_Wh=0.0, area_name=area_name,
+                 time_slot=pendulum.datetime(2022, 2, 1, 22, 15)),
+            call(purchased_energy_Wh=0.0, area_name=area_name,
+                 time_slot=pendulum.datetime(2022, 2, 1, 22, 30)),
+            call(purchased_energy_Wh=0.0, area_name=area_name,
+                 time_slot=pendulum.datetime(2022, 2, 1, 22, 45)),
+            call(purchased_energy_Wh=0.0, area_name=area_name,
+                 time_slot=pendulum.datetime(2022, 2, 1, 23, 0)),
+            call(purchased_energy_Wh=0.0, area_name=area_name,
+                 time_slot=pendulum.datetime(2022, 2, 1, 23, 15)),
+            call(purchased_energy_Wh=0.0, area_name=area_name,
+                 time_slot=pendulum.datetime(2022, 2, 1, 23, 30)),
+            call(purchased_energy_Wh=0.0, area_name=area_name,
+                 time_slot=pendulum.datetime(2022, 2, 1, 23, 45))
+        ]
+
+        state_mock.decrement_energy_requirement.assert_has_calls(calls)
+
+
+@pytest.fixture(name="pv")
+def pv_fixture():
+    """PV asset."""
+    return Area(
+        "PV", strategy=PVStrategy(2, 80)
+    )
+
+
+class TestProductionStandardProfileEnergyParameters:
+    """Tests for the ConsumptionStandardProfileEnergyParameters class."""
+
+    @staticmethod
+    def test_event_traded_energy(pv):
+        energy_params = ProductionStandardProfileEnergyParameters(capacity_kW=200)
+        state_mock = MagicMock()
+        energy_params._state = state_mock  # pylint: disable=protected-access
+        # We call this method to set the area and the profile generator
+        energy_params.event_activate_energy(pv)
+
+        market_slot = pendulum.datetime(2022, 2, 1)
+        product_type = AvailableMarketTypes.DAY_FORWARD
+        energy_params.event_traded_energy(
+            energy_kWh=50,
+            market_slot=market_slot,
+            product_type=product_type)
+
+        assert state_mock.decrement_available_energy.call_count == 96
+
+        area_name = "PV"
+        # We want to make sure that all the expected calls to the state class are correctly issued
+        calls = [
```

Comment:
What kind of `for` loop would you recommend using? Each call is different: the first N calls have a different datetime but always `0` energy. The following calls (the middle of the profile) all have different energy values. Finally, the tail of the profile again has the same energy (`0`) but different datetimes.

I could in theory create a list of dicts for each call (like `[{date: ..., energy: ...}, {date: ..., energy: ...} ...]` and the use a `for` loop to create the `call` objects, but IMO it wouldn't change much. So I think this is not what you were thinking of.

Reply:
I got it. However, the energy value seems to be a normal curve and I was thinking of generating the curve at run time instead of hard coding it there. Is it possible (easy) to implement?

Reply:
AFAIU, the energy values are from standard_solar_profile.csv
If we want to change the file, we need to also change every line here. So yeah, a list comprehension that reads the SSP for the selected time_slot could work. However, then the test is influenced by the CSV parser, which is not what we want right?

Reply:
@hannesdiedrich exactly. Basically we don't want to implement logic in tests, because that logic should then be tested as well. Specifically: if we create a function that generates a gaussian curve (that replaces the SSP) we would then also need to be sure that that function works properly. So the straightforward way is to just hardcode the values we expect to see as an output for the current query we're testing. If the SSP changes, tests will fail. This might or might not be desired, but it's better than testing something else by mistake and seeing tests passing by accident (without catching bugs).